src/pages/authentication/Login.jsx

```diff
@@ -0,0 +1,149 @@
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { useAuthContext } from "../../context/AuthContext";
+import { PostApi } from "../../apicalls/PostApi";
+
+const inititalLoginState = { username: "", password: "" };
+
+const validateForm = (formState) => {
+  const { username, password } = formState;
+
+  const errors = {};
+
+  if (!username) {
+    errors.username = "username is required!";
+  }
+  if (!password) {
+    errors.password = "Password is required";
+  }
+  return errors;
+};
+
+const Login = () => {
+  const [showPassword, setShowPassword] = useState(false);
+  const [loginFormState, setLoginFormState] = useState(inititalLoginState);
+  const [formError, setFormError] = useState(inititalLoginState);
+  const { authDispatch } = useAuthContext();
+  const location = useLocation();
+  const navigate = useNavigate();
+
+  const loginUserApiCall = async () => {
+    const response = await PostApi(
+      "/api/auth/login",
+      { ...loginFormState },
+      false
+    );
+    const { data, success } = response;
+    if (success) {
+      authDispatch({ type: "SET_LOGGED_USER", payload: data });
+      location.state === null
+        ? navigate("/scroll")
+        : navigate(location?.state?.from);
+    } else {
+      alert("Something went wrong, check console");
+    }
+  };
+
+  const loginUser = (e) => {
+    e.preventDefault();
+    const errors = validateForm(loginFormState);
+    if (Object.keys(errors).length === 0) {
+      setFormError(inititalLoginState);
+      loginUserApiCall();
+    } else {
+      setFormError(errors);
+    }
+  };
+  const inputChangeHandler = (e) => {
+    setFormError(inititalLoginState);
```

Comment:
suggestion: you can write this line after submitting the form, instead of doing at every input change event.

Reply:
Done